okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/FastFallbackExchangeFinder.kt

```diff
@@ -35,132 +35,139 @@ internal class FastFallbackExchangeFinder(
 ) {
   private val connectDelayMillis = 250L
 
-  /** Plans currently being connected, and that will later be added to [connectResults]. */
-  private var connectsInFlight = CopyOnWriteArrayList<Plan>()
+  /**
+   * Plans currently being connected, and that will later be added to [connectResults]. This is
+   * mutated by the call thread only. If is accessed by background connect threads.
+   */
+  private val tcpConnectsInFlight = CopyOnWriteArrayList<Plan>()
+
+  /** Plans we started, and canceled, because another plan won a race. */
```

Comment:
Maybe worth a comment that this is always a clean plan, cancelled and retryed.

The comment here suggests it's almost the same plan.

```
      plan.cancel()
      val retry = plan.retry() ?: continue
```

Reply:
Or is that a feature of the current strategy, and potentially these could be in progress ones, that we are holding for some time?

Reply:
Yeah this doc is wrong. I changed to this:
```
  /**
   * These are retries of plans that were canceled when they lost a race. If the race's winner ends
   * up not working out, this is what we'll attempt first.
   */
```